docs/readme/README.jl

```diff
@@ -0,0 +1,247 @@
+# # DataFlowTasks.jl
+#
+#md # [![Stable](https://img.shields.io/badge/docs-stable-blue.svg)](https://maltezfaria.github.io/DataFlowTasks.jl/stable)
+#md # [![Dev](https://img.shields.io/badge/docs-dev-blue.svg)](https://maltezfaria.github.io/DataFlowTasks.jl/dev)
+#md # [![Build Status](https://github.com/maltezfaria/DataFlowTasks.jl/workflows/CI/badge.svg)](https://github.com/maltezfaria/DataFlowTasks.jl/actions)
+#md # [![codecov](https://codecov.io/gh/maltezfaria/DataFlowTasks.jl/branch/main/graph/badge.svg?token=UOWU691WWG)](https://codecov.io/gh/maltezfaria/DataFlowTasks.jl)
+#md # ![Lifecycle](https://img.shields.io/badge/lifecycle-experimental-blue.svg)
+#
+# `DataFlowTasks.jl` is a Julia package dedicated to parallel programming on multi-core shared memory CPUs. From user annotations (READ, WRITE, READWRITE) on program data, `DataFlowTasks.jl` automatically infers dependencies between parallel tasks.
+
+#md # This `README` is also available in notebook form:
+#md # [![ipynb](https://img.shields.io/badge/download-ipynb-blue)](docs/readme/README.ipynb)
+#md # [![nbviewer](https://img.shields.io/badge/show-nbviewer-blue.svg)](https://nbviewer.jupyter.org/github/maltezfaria/DataFlowTasks.jl/blob/main/docs/readme/README.ipynb)
+
+#-
+
+# ## Installation
+#
+# ```julia
+# using Pkg
+# Pkd.add("https://github.com/maltezfaria/DataFlowTasks.jl.git")
+# ```
+
+#-
+
+# ## Basic Usage
+#
+# `DataFlowTasks.jl` allows decomposing a program into tasks, that are meant to be used in a very similar way to Julia native `Task`s. The API implements three macros :
+#
+# - `@dspawn`
+# - `@dtask`
+# - `@dasync`
+#
+# which behave like their `Base` counterparts, except they take additional annotations that declare how each task affects the data it accesses:
+#
+# - read-only: `@R` or `@READ`
+# - write-only: `@W` or `@WRITE`
+# - read-write: `@RW` or `@READWRITE`
+#
+# Anywhere in the task body, a `@R(A)` annotation for example implies that `A` will be accessed in read-only mode by the task.
+
+#-
+
+# Let's look at a simple example:
+
+using DataFlowTasks
+DataFlowTasks.reset!() #hide
```

Comment:
Unresolved questions regarding `reset`:
- do we want to keep it? (cf. #24)
- even if so, do we want to advertize its use?

OTOH: if we don't want to have/advertize it, make sure removing it doesn't affect the outcome of the examples shown here.

Reply:
Oh, and if we want to keep `reset()`, we'll have to add tests for it.